### PR TITLE
WLA-442: Steps Batch 3

### PR DESCRIPTION
**Pull request by LucaDelBuonoHMCTS**

### CODE REVIEW

File: steps/identity/appellant-contact-details/answer.html [hunk 1]
```diff
@@ -1,34 +1,32 @@
-
+<div class="govuk-summary-list">
+    <dl class="govuk-summary-list__row">
+        <dt class="govuk-summary-list__key">{{ content.cya.address.question | safe }}</dt>
+        <dd class="govuk-summary-list__value">
+            {{ fields.addressLine1.value | safe }}<br>
```
Comment (timja) [inline steps/identity/appellant-contact-details/answer.html:5]: This looks like a potential XSS vulnerability where HTML can be injected into the page,
has this been sanitised elsewhere?

What's the reason for this change?

Comment (LucaDelBuonoHMCTS) [reply steps/identity/appellant-contact-details/answer.html:5]: You mean adding the ` | safe` filter?

Comment (timja) [reply steps/identity/appellant-contact-details/answer.html:5]: Yes safe allows adding arbitrary html and should never be used on unsanitised user input, and should really just not be used on user input in general

Comment (LucaDelBuonoHMCTS) [reply steps/identity/appellant-contact-details/answer.html:5]: You're right, I had to add it to thousands of fields for Welsh and couldn't really check where every single one of them came from.
I guess I'll have to do a further scan to find them all.

File: steps/reasons-for-appealing/reason-for-appealing/answer.html [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
-                </dd>
-                <dd class="govuk-summary-list__actions"></dd>
-            </dl>
-        </div>
+    <div class="govuk-summary-list">
+        <dl id="cya-reasonforappealing-reason-for-appealing-{{ loop.index }}" class="govuk-summary-list__row">
+            <dt class="govuk-summary-list__key">{{ content.cya.reasonForAppealing.questions.whatYouDisagreeWith | safe }}</dt>
+            <dd class="govuk-summary-list__value">
+                {% if item.whatYouDisagreeWith and item.whatYouDisagreeWith !== ' ' %}
+                    {{ item.whatYouDisagreeWith | safe }}
```
Comment (timja) [inline steps/reasons-for-appealing/reason-for-appealing/answer.html:7]: is this provided by the user?

File: steps/reasons-for-appealing/reason-for-appealing/answer.html [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
+                <span class="govuk-visually-hidden">{{ content.cya.reasonForAppealing.reason | safe }} {{ loop.index }}</span>
+            </dd>
+        </dl>
+    </div>
+    <div class="govuk-summary-list">
+        <dl class="govuk-summary-list__row">
+            <dt class="govuk-summary-list__key">{{ content.cya.reasonForAppealing.questions.reasonForAppealing | safe }}</dt>
+            <dd class="govuk-summary-list__value">
+                {% if item.reasonForAppealing and item.reasonForAppealing !== ' ' %}
+                    {{ item.reasonForAppealing | safe }}
```
Comment (timja) [inline steps/reasons-for-appealing/reason-for-appealing/answer.html:23]: is this provided by the user?

File: steps/reasons-for-appealing/reason-for-appealing/template.html [hunk 4]
```diff
@@ -32,47 +32,44 @@
...
 
-        {{ content.noReasons }}
-
+        {{ content.noReasons | safe }}
     </div>
 {% endblock %}
 
 {% block item %}
-    {{ item.whatYouDisagreeWith.value }}
+    {{ item.whatYouDisagreeWith.value | safe }}
```
Comment (timja) [inline steps/reasons-for-appealing/reason-for-appealing/template.html:56]: is this provided by the user?

File: steps/identity/appellant-contact-details/answer.html [hunk 5]
```diff
@@ -1,34 +1,32 @@
-
+<div class="govuk-summary-list">
+    <dl class="govuk-summary-list__row">
+        <dt class="govuk-summary-list__key">{{ content.cya.address.question | safe }}</dt>
```
Comment (timja) [inline steps/identity/appellant-contact-details/answer.html:3]: why do you need to add safe at all? it should only be needed for embedded html and that should only be done on a case by case basis

Comment (LucaDelBuonoHMCTS) [reply steps/identity/appellant-contact-details/answer.html:3]: In case characters are encoded (that was the case in Probate for example).
